website/docs/developer/research/research-orca-delete-functionality.md

```diff
@@ -0,0 +1,122 @@
+
+- As a DAAC, I would like duplicate data to be versioned in ORCA in order to recover the proper version of the data based on the type of failure.	
+
+
+- Perform a check with CMR and/or Cumulus to verify that the granule has been deleted from those locations prior to allowing the delete to proceed in ORCA.
+
+    
+    This can be implemented in the existing lambda function that will be used to delete the data from ORCA.
+
+### API gateway
+
+API gateway is currently used by end users for creating reconciliation reports and getting status update of recovery granules and jobs. This can be used for creating API calls made by end users for deletion. The API gateway should act as a trigger to the lambda functions that actually perform the deletion in the backend. An API gateway with two different resources need to be created- one for the soft delete API call and one for the hard delete API call.
+Check API gateway [examples](https://github.com/nasa/cumulus-orca/tree/develop/modules/api-gateway) currently used by ORCA. 
+
+### Lambda function
+
+The delete functionality can be performed by two lambda functions based on initial research. One of the lambda functions will be used to perform soft delete- delete from S3 bucket but only flag the granule in the database for deletion after 90 days. The lambda should also verify with CMR and/or Cumulus that the granule has been deleted from those locations prior to allowing the delete to proceed in ORCA. The lambda will be triggered by an API call made by the user once the user is authenticated with Launchpad.
+
+The other lambda function can be used to perform a physical delete- delete from S3 glacier as well as from the database. Similar to soft delete, it should first verify with CMR and/or Cumulus that the granule has been deleted from those locations before deleting from ORCA. This will require further discussion with Cumulus team before performing the actual work.
+
+
+### Deleting objects from AWS S3
+
+To delete an object from an S3 bucket, the [delete_objects](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.delete_objects)
+can be used from the boto3 library. An example of the usage can be seen below.
+Some notable items if this library is used includes the following.
+- It can be used to delete up to 1000 objects from a bucket using a single API call.
+- When performing this action on an MFA Delete enabled bucket that attempts to delete any versioned objects, an MFA token must be included orelse it will fail.
+- If the object specified in the request is not found, Amazon S3 returns the result as deleted.
+- Each DeleteObjects API incurs a little cost.
```

Comment:
What is the cost? When removing versioned data granule, do we remove all the versions or do we remove all but the last and keep the delete marked version for longer?

Reply:
looks like cost is $0.05/10000 requests according to according to https://www.sumologic.com/insight/s3-cost-optimization

Regarding the other question, I think that should be discussed with the Cumulus or larger team to get insights. Added to "questions to be asked" section.